lib/rubocop/cop/lint/useless_trailing_comma.rb

```diff
@@ -0,0 +1,113 @@
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks whether trailing commas in block arguments are
+      # required. Blocks with only one argument and a trailing comma require
+      # that comma to be present. Blocks with more than one argument never
+      # require a trailing comma.
+      #
+      # @example
+      #   # bad
+      #   add { |foo, bar,| foo + bar }
+      #
+      #    # good
+      #   add { |foo, bar| foo + bar }
+      #
+      #   # good
+      #   add { |foo,| foo }
+      #
+      #   # good
+      #   add { foo }
+      #
+      #   # bad
+      #  add do |foo, bar,|
+      #    foo + bar
+      #   end
+      #
+      #  # good
+      #  add do |foo, bar|
+      #    foo + bar
+      #  end
+      #
+      #  # good
+      #   add do |foo,|
+      #    foo
+      #  end
+      #
+      #  # good
+      #  add do
+      #     foo + bar
+      #  end
+      class UselessTrailingComma < Cop
+        include RangeHelp
+
+        MSG = 'Block arguments do not require a trailing comma except when ' \
+              'only a single argument is present.'
+
+        ARG_MSG = 'Useless trailing comma present in block arguments.'
+
+        def on_block(node)
+          return unless useless_trailing_comma?(node)
+
+          add_offense_for_args(node, node.arguments, ARG_MSG)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            return unless useless_trailing_comma?(node)
+
+            end_pos = range_with_surrounding_space(
+              range: node.arguments.source_range,
+              side: :right,
+              newlines: false
+            ).end_pos
+            range = range_between(node.loc.begin.end.begin_pos, end_pos)
+            corrector.replace(range, block_arg_string(node, node.arguments))
+          end
+        end
+
+        private
+
+        def block_arg_string(node, args)
+          arg_string = args.children.map do |arg|
+            if arg.mlhs_type?
+              "(#{block_arg_string(node, arg)})"
+            else
+              arg.source
+            end
+          end.join(', ')
+          arg_string = " |#{arg_string}|"
+          arg_string += ' ' if single_line_block?(node)
+          arg_string
+        end
+
+        def add_offense_for_args(node, args, msg)
+          arguments = args.source_range
+          range = range_between(arguments.begin_pos, arguments.end_pos)
+
+          add_offense(node, location: range, message: msg)
+        end
+
+        def useless_trailing_comma?(node)
+          arg_count(node) > 1 && trailing_comma?(node)
+        end
+
+        def arg_count(node)
+          %i[arg optarg kwoptarg].map do |arg_type|
+            node.arguments.each_descendant(arg_type).to_a.size
+          end.reduce(:+)
+        end
+
+        def single_line_block?(node)
+          !node.source.include?("\n")
+        end
+
+        def trailing_comma?(node)
+          node.arguments.source.count(',') == arg_count(node)
```

Comment:
Would it be more robust to check that the last non-whitespace character is `,`?

Right now, I suspect this will return `true` for e.g. an empty block:

```ruby
lambda do ||
end
```